hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java

```diff
@@ -172,17 +168,29 @@ public Data txnCommitPollBackup(long itemId) {
     }
 
     public boolean txnRollbackPoll(long itemId, boolean backup) {
-        QueueItem item = txMap.remove(itemId);
+        TxQueueItem item = txMap.remove(itemId);
         if (item == null) {
             return false;
         }
         if (!backup) {
-            getItemQueue().offerFirst(item);
+            addTxItemOrdered(item);
         }
         cancelEvictionIfExists();
         return true;
     }
 
+    private void addTxItemOrdered(TxQueueItem txQueueItem) {
+        ListIterator<QueueItem> iterator = ((List) getItemQueue()).listIterator();
```

Comment:
we have to use `getItemQueue()` method because there is no queue created for backup but will be created in case of promote via that method. And check-style is bitching about returning `LinkedList`